Move requirements for packet number reset (#3811)
quicwg/base-drafts: draft-ietf-quic-transport.md

After a client receives a Retry packet, 0-RTT packets are likely to have been
lost or discarded by the server.  A client SHOULD attempt to resend data in

> 0-RTT packets after it sends a new Initial packet.  New packet numbers MUST be

Reviewer: ```suggestion
0-RTT packets after it sends a new Initial packet.  New larger packet numbers MUST be
```

Reviewer: I wasn't 100% sure what new meant, so I think adding larger avoids any potential confusion.

Author: I think that the "reusing packet numbers" is clear enough.

> used for any new packets that are sent; as defined in {{retry-continue}},

Reviewer 2: ```suggestion
used for any new packets that are sent; as described in {{retry-continue}},
```

the cryptographic handshake message it sends in response to receiving a Retry.

A client MUST NOT reset the packet number for any packet number space after

> processing a Retry packet. This applies to 0-RTT packets in particular as those
> contain confidential information that is most likely to need transmission.  The
> keys used to protect packets will not change as a result of responding to a
> Retry packet.  Sending packets with the same packet number is likely to
> compromise the packet protection for those packets because the same key and
> nonce could be used to protect different content.

Reviewer 2: ```suggestion
processing a Retry packet. In particular, 0-RTT packets contain confidential
information that will most likely be retransmitted on receiving a Retry packet.
The keys used to protect these new 0-RTT packets will not change as a result of
responding to a Retry packet. However, the data sent in these packets could be
different than what was sent earlier. Sending these new packets with the same
packet number is likely to compromise the packet protection for those packets
because the same key and nonce could be used to protect different content.
```
